packages/core-js-sdk/test/createSdk.test.ts

```diff
@@ -0,0 +1,35 @@
+import createSdk from '../src/index';
+
+const mockFetch = jest.fn();
+globalThis.fetch = mockFetch;
+
+describe('createSdk', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call "beforeRequest" and "afterRequest"', async () => {
```

Comment:
I would split it into 2 different tests and test that when changing the hook it's actually working
I think that we already have tests for calling hooks

Reply:
unfortunately we don't have tests for this code
(only with mocks of the inner object)

Reply:
good idea for splitting, done 👍

Reply:
Cool, thanks!
Don't you want to add a test where the hooks are updated after initialization?
just to make sure someone will not change it without noticing

Reply:
good idea 👍  added such test